File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
            map<int,int> m1;
        vector<int> ans(2);
        for(int i=0;i<nums.size();i++){
            if(m1.find(target-nums[i])!=m1.end()){
                ans[0]=m1[target-nums[i]];
                ans[1]=i;
                return ans;
            }
            m1[nums[i]]=i;
        }
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = new int[2];
        Map<Integer, Integer> m1= new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(m1.containsKey(target - nums[i])){
                ans[0] = m1.get(target - nums[i]);
                ans[1] = i;
                return ans;
            }
            m1.put(nums[i],i);
        }
        return ans;

    }
}